clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -402,27 +403,41 @@ public void refreshCommittedOffsetsIfNeeded() {
         }
     }
 
-    @Override
-    public void close() {
+    public void close(long timeoutMs, long requestTimeoutMs) {
         // we do not need to re-enable wakeups since we are closing already
         client.disableWakeups();
+
+        long now = time.milliseconds();
+        timeoutMs = Math.min(timeoutMs, requestTimeoutMs);
+        long endTimeMs = now + timeoutMs;
         try {
-            maybeAutoCommitOffsetsSync();
-
-            Node coordinator;
-            long endTimeMs = time.milliseconds() + CLOSE_TIMEOUT_MS;
-            while ((coordinator = coordinator()) != null && client.pendingRequestCount(coordinator) > 0) {
-                long remainingTimeMs = endTimeMs - time.milliseconds();
-                if (remainingTimeMs > 0)
-                    client.poll(remainingTimeMs);
-                else {
-                    log.warn("Close timed out with {} pending requests to coordinator, terminating client connections for group {}.", client.pendingRequestCount(coordinator), groupId);
-                    break;
-                }
+            maybeAutoCommitOffsetsSync(timeoutMs);
+            now = time.milliseconds();
+            if (pendingAsyncCommits.get() > 0) {
+                ensureCoordinatorReady(now, endTimeMs - now);
+                now = time.milliseconds();
             }
         } finally {
             super.close();
         }
+
+        // At this point, there may be pending commits (async commits or sync commits that were
```

Comment:
Done. The default `AbstractCoordinator.close()` used by `WorkerCoordinator` uses  a timeout of zero (attempt to send LeaveGroup without waiting for response).